website/tokens/__init__.py

```diff
@@ -0,0 +1,95 @@
+import httplib as http
+import functools
+import jwt
+from flask import request
+
+from framework.exceptions import HTTPError
+
+from website import settings
+from website.tokens import handlers
+from website.tokens.exceptions import TokenHandlerNotFound
+
+class TokenHandler(object):
+
+    HANDLERS = {
+        'approve_registration_approval': functools.partial(handlers.sanction_handler, 'registration', 'approve'),
+        'reject_registration_approval': functools.partial(handlers.sanction_handler, 'registration', 'reject'),
+        'approve_embargo': functools.partial(handlers.sanction_handler, 'embargo', 'approve'),
+        'reject_embargo': functools.partial(handlers.sanction_handler, 'embargo', 'reject'),
+        'approve_retraction': functools.partial(handlers.sanction_handler, 'retraction', 'approve'),
+        'reject_retraction': functools.partial(handlers.sanction_handler, 'retraction', 'reject')
+    }
+
+    def __init__(self, encoded_token=None, payload=None):
+
+        self.encoded_token = encoded_token
+        self.payload = payload
+
+    @staticmethod
+    def encode(payload):
+        return jwt.encode(
+            payload,
+            settings.JWT_SECRET,
+            algorithm=settings.JWT_ALGORITHM
+        )
+
+    @staticmethod
+    def decode(encoded_token):
+        return jwt.decode(
+            encoded_token,
+            settings.JWT_SECRET,
+            algorithms=[settings.JWT_ALGORITHM])
+
+    @classmethod
+    def from_string(cls, encoded_token):
+        try:
+            payload = TokenHandler.decode(encoded_token)
+        except jwt.DecodeError as e:
+            raise HTTPError(
+                http.BAD_REQUEST,
+                data={
+                    'message_short': 'Bad request',
+                    'message_long': e.message
+                }
+            )
+        return cls(encoded_token=encoded_token, payload=payload)
+
+    @classmethod
+    def from_payload(cls, payload):
+        encoded_token = TokenHandler.encode(payload)
+        return cls(encoded_token=encoded_token, payload=payload)
+
+    def to_response(self):
+        action = self.payload.get('action', None)
+        handler = self.HANDLERS.get(action)
+        if handler:
+            return handler(self.payload, self.encoded_token)
+        else:
+            raise TokenHandlerNotFound(action=action)
+
+def process_token_or_pass(func):
```

Comment:
@leb2dg, this will change the way you do DraftRegistration approval/rejection. 

No action needed, but when you come back we should discuss the implications of this.